p2p/tests/sessions.rs

```diff
@@ -217,46 +209,103 @@ fn p2p_sessions_is_outbound_bootstrap_needed() {
     assert!(!sessions.is_outbound_bootstrap_needed());
 }
 
-/// Check the registration of sessions
+/// Check the function to get a random outbound consolidated session
 #[test]
-fn p2p_sessions_register() {
+fn p2p_sessions_get_random_anycast_session() {
     // Create sessions struct
     let mut sessions = Sessions::<String>::default();
 
+    // Check that the function returns None when there are no sessions in the collection
+    assert_eq!(sessions.get_random_anycast_session(), None);
+
     // Register an outbound session and check if result is Ok(())
+    let reference1 = "reference1".to_string();
     let outbound_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 8001);
-    assert!(
-        sessions
-            .register_session(
-                SessionType::Outbound,
-                outbound_address,
-                "reference1".to_string()
-            )
-            .is_ok()
+    assert!(sessions
+        .register_session(SessionType::Outbound, outbound_address, reference1.clone())
+        .is_ok());
+
+    // Check that the function returns None when there are no consolidated sessions in the
+    // collection
+    assert_eq!(sessions.get_random_anycast_session(), None);
+
+    // Consolidate outbound session
+    assert!(sessions
+        .consolidate_session(SessionType::Outbound, outbound_address)
+        .is_ok());
+
+    // Check that the function returns Some(T) when there is one valid session in the collection
+    assert_eq!(
+        sessions.get_random_anycast_session(),
+        Some("reference1".to_string())
     );
 
-    // Check if outbound session was registered successfully into the unconsolidated sessions
+    // Register and consolidate an outbound session and check if result is Ok(())
+    let reference2 = "reference2".to_string();
+    let outbound_address = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 8002);
+    assert!(sessions
+        .register_session(SessionType::Outbound, outbound_address, reference2.clone())
+        .is_ok());
+    assert!(sessions
+        .consolidate_session(SessionType::Outbound, outbound_address)
+        .is_ok());
+
+    // Get random session for a "big" number
+    let mut diff: i16 = 0;
+    for _ in 0..100000 {
+        // Get a random anycast sessions (there are only 2)
+        match sessions.get_random_anycast_session() {
```

Comment:
does `&sessions.get_random_anycast_session()` help here instead of clone later?

Reply:
Sure, thanks!